packages/core-mobile/app/contexts/SwapContext/performSwap/performSwap.ts

```diff
@@ -77,6 +75,13 @@ export async function performSwap({
 
   // no need to approve native token
   if (!isSrcTokenNative) {
+    const [spenderAddress, spenderAddressError] = await resolve<string>(
+      SwapService.getParaswapSpender(activeNetwork) ?? Promise.resolve(null)
+    )
+    if (spenderAddressError || spenderAddress === null) {
+      throw new Error(`Spender Address Error: ${spenderAddressError}`)
+    }
+
```

Comment:
I guess `await resolve` were used below because `contract.allowance`, `contract.approve` can be undefined, but here, I think we just can call `await SwapService.getParaswapSpender` with try catch.